environments/key4hep-desy-release/desy-packages.yaml

```diff
@@ -0,0 +1,52 @@
+packages:
+  all:
+    target: [x86_64]
+    variants: build_type=RelWithDebInfo cxxstd=17
+    buildable: true
+    version: []
+    providers:
+      gl: [mesa]
+      glu: [mesa]
+    compiler: [gcc@11.2.0]
+
+  # Key4hep and ilcsoft packages
+  marlin:
+    variants: +lccd
+  dd4hep:
+    variants: +lcio+xercesc+hepmc3+edm4hep
+  podio:
+    variants: +sio
+
+  # "Core external" packages
+  geant4:
+    variants: ~qt+opengl+vecgeom cxxstd=17
+    version: [11.0.1]
```

Comment:
Note that unlike the old concretizer, the clingo concretizer treats packages.yaml as a "soft preference" and will change versions if there is a conflict for example. To get a hard requirement, add `geant4@11.0.1` to the specs to be installed in the environment and concretize together.

Reply:
I have been wondering about that actually in some other cases. That is good to know.